hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -401,6 +403,45 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		CloudResourcesDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+		if CloudResourcesDestroyedCondition.Status == "False" {
+			if hcluster.Spec.Platform.Type != hyperv1.NonePlatform &&
+				hcluster.Spec.Platform.Type != hyperv1.AgentPlatform &&
+				hcluster.Spec.Platform.Type != hyperv1.KubevirtPlatform {
+
+				var ConditionCreationTime time.Time
+				var timeLimit time.Time
+
+				reportingGracePeriod := &metav1.Condition{
+					Type:               string(hyperv1.HostedClusterDestroyed),
+					Status:             metav1.ConditionTrue,
+					Reason:             hyperv1.Done,
+					LastTransitionTime: metav1.Now(),
+				}
+
+				HostedClusterDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.HostedClusterDestroyed))
+				if HostedClusterDestroyedCondition != nil {
+					reportingGracePeriod = HostedClusterDestroyedCondition
+				} else {
+					ConditionCreationTime = time.Now()
+					timeLimit = ConditionCreationTime.Add(reportingGracePeriodDuration)
+				}
+
+				reportingGracePeriod.Message = fmt.Sprintf("Creation: %v, Limit: %v Resources: %v", ConditionCreationTime, timeLimit, CloudResourcesDestroyedCondition.Message)
+
+				meta.SetStatusCondition(&hcluster.Status.Conditions, *reportingGracePeriod)
+				if err := r.Client.Status().Update(ctx, hcluster); err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
+				}
+
+				if reportingGracePeriod.LastTransitionTime.Time.Before(timeLimit) {
```

Comment:
In case it reconciles and the "HostedClusterDestroyedCondition" exists, isn't the timeLimit empty?

Reply:
Yep, that was an early version that still was in testing, fixed now